File: yh_connect/scripts/yh_connect_sub_pub.py
#!/usr/bin/python
#-*- coding: utf-8 -*-  
                               
import rospy
from std_msgs.msg import Int32, Float32

class ConnectSubPub:
    def __init__(self):
        self.sub = rospy.Subscriber("yh_topic",Int32,self.msgCallback)
        self.pub = rospy.Publisher("yh_topic2",Float32)
        self.float_msg = Float32()

    def msgCallback(self,msg):
        if msg.data % 5 == 0:
            self.float_msg.data = msg.data/3
            self.pub.publish(self.float_msg)

if __name__ ==  "__main__":
    rospy.init_node("yh_connect_sub_pub")
    connect_sub_pub = ConnectSubPub() 
    rospy.spin()

File: yh_connect/src/yh_connect_sub_pub.cpp
#include "ros/ros.h"
#include "std_msgs/Int64.h"
#include "std_msgs/Float64.h"

ros::Publisher pub;

void msgCallback(const std_msgs::Int64::ConstPtr& msg)
{
    if (msg->data %5 == 0)
    {
        std_msgs::Float64 float_msg;
        float_msg.data = (float)msg->data/3.0f;  //(float)을 써주면 int msg.data가 float으로 바뀐다.
        pub.publish(float_msg); 
        ROS_INFO("sibal~");
    }
   
}
int main(int argc, char** argv)
{
    ros::init(argc,argv, "yh_connect_sub_pub");
    ros::NodeHandle nh;

    ros::Subscriber sub = nh.subscribe("yh_topic",30,msgCallback);
    pub = nh.advertise<std_msgs::Float64>("yh_topic2",10);

    ros::spin(); 
    
    return 0;
}